File: src/main/java/coza/trojanc/receipt/loader/Loader.java
package coza.trojanc.receipt.loader;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

/**
 * @author Charl Thiem
 */
public interface Loader<T> {

	T load(InputStream inputStream) throws IOException;

	T load(String jsonString) throws IOException;

	void write(T instance, OutputStream out) throws IOException;
}

File: receipt-java/src/main/kotlin/coza/trojanc/receipt/loader/Loader.kt
package coza.trojanc.receipt.loader

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream

/**
 * @author Charl Thiem
 */
interface Loader<T> {

    @Throws(IOException::class)
    fun load(inputStream: InputStream): T

    @Throws(IOException::class)
    fun load(jsonString: String): T

    @Throws(IOException::class)
    fun write(instance: T, out: OutputStream)
}
